main/document/add_link.php

```diff
@@ -0,0 +1,184 @@
+    }
+}
+ 
+$is_allowed_to_edit = api_is_allowed_to_edit(null, true);
+ 
+if (api_get_group_id()) {
+    // If the group id is set, check if the user has the right to be here
+    // Needed for group related stuff
+    require_once api_get_path(LIBRARY_PATH).'groupmanager.lib.php';
+    // Get group info
+    $group_properties = GroupManager::get_group_properties(api_get_group_id());
+
+    if ($is_allowed_to_edit || GroupManager::is_user_in_group($_user['user_id'], api_get_group_id())) { // Only courseadmin or group members allowed
+        $to_group_id = api_get_group_id();
+        $req_gid = '&amp;gidReq='.api_get_group_id();
+        $interbreadcrumb[] = array('url' => '../group/group_space.php?gidReq='.api_get_group_id(), 'name' => get_lang('GroupSpace'));
+    } else {
+        api_not_allowed(true);
+    }
+} elseif ($is_allowed_to_edit || DocumentManager::is_my_shared_folder(api_get_user_id(), $path, api_get_session_id())) {
+    // Admin for "regular" upload, no group documents. And check if is my shared folder
+    $to_group_id = 0;
+    $req_gid = '';
+} else { // No course admin and no group member...
+    api_not_allowed(true);
+}
+ 
+// Group docs can only be uploaded in the group directory
+if ($to_group_id != 0 && $path == '/') {
+    $path = $group_properties['directory'] . "/";
+}
+ 
+// Breadcrumbs
+if ($is_certificate_mode) {
+    $interbreadcrumb[] = array('url' => '../gradebook/'.$_SESSION['gradebook_dest'], 'name' => get_lang('Gradebook'));
+} else {
+    $interbreadcrumb[] = array('url' => './document.php?id='.$document_id.$req_gid, 'name'=> get_lang('Documents'));
+}
+ 
+// Interbreadcrumb for the current directory root path
+if (empty($document_data['parents'])) {
+    // Hack in order to not add the document to the breadcrumb in case it is a link
+    if ($document_data['filetype'] != 'link') {
+        $interbreadcrumb[] = array('url' => '#', 'name' => $document_data['title']);
+    }
+} else {
+    foreach ($document_data['parents'] as $document_sub_data) {
+        // Hack in order to not add the document to the breadcrumb in case it is a link
+        if ($document_data['filetype'] != 'link') {
+            $interbreadcrumb[] = array('url' => $document_sub_data['document_url'], 'name' => $document_sub_data['title']);
+        }
+    }
+}
+ 
+$this_section = SECTION_COURSES;
+ 
+$nameTools = get_lang('LinkAdd');
+
+// Display the header
+Display::display_header($nameTools, 'Doc');
+ 
+/*	Here we do all the work */
+ 
+// Actions
+echo '<div class="actions">';
+// Link back to the documents overview
+if ($is_certificate_mode) {
+    echo '<a href="document.php?id='.$document_id.'&selectcat=' . $selectcat.'">'.Display::return_icon('back.png', get_lang('BackTo').' '.get_lang('CertificateOverview'), '', ICON_SIZE_MEDIUM).'</a>';
+} else {
+    echo '<a href="document.php?id='.$document_id.'">'.Display::return_icon('back.png', get_lang('BackTo').' '.get_lang('DocumentsOverview'), '', ICON_SIZE_MEDIUM).'</a>';
+}

```

Comment:
This if() could then be removed as well, focusing on the case where $is_certificate_mode == false